### PR TITLE
[GNA] Add missing metrics to GNAExecutableNetwork

### PR DESCRIPTION
**Pull request by rjeziers**
### Details:
Added following metrics to GNAExecutableNetwork:
- NETWORK_NAME

### Tickets:
 - 18180

### CODE REVIEW

**Review by sirzabek — COMMENTED** [review]
Are you planning to add some tests for the extended metrics set?

**Comment by rjeziers** [comment]
I'll add some unit tests for this

File: src/plugins/intel_gna/src/gna_executable_network.hpp [hunk 1]
```diff
@@ -128,8 +131,16 @@ class GNAExecutableNetwork : public InferenceEngine::IExecutableNetworkInternal
     }
 
     InferenceEngine::Parameter GetMetric(const std::string& name) const override {
-        if (ov::supported_properties == name) {
+        if (ov::supported_properties == name || InferenceEngine::Metrics::METRIC_SUPPORTED_CONFIG_KEYS == name) {
```
Comment (nkogteva) [inline src/plugins/intel_gna/src/gna_executable_network.hpp:134]: We don't need add support of legacy API metrics.

Comment (rjeziers) [reply src/plugins/intel_gna/src/gna_executable_network.hpp:134]: By legacy API, do you mean InferenceEngine::Metrics::METRIC_SUPPORTED_CONFIG_KEYS and InferenceEngine::Metrics::METRIC_SUPPORTED_METRICS?
So only ov::model_name is necessary?

Comment (nkogteva) [reply src/plugins/intel_gna/src/gna_executable_network.hpp:134]: > By legacy API, do you mean InferenceEngine::Metrics::METRIC_SUPPORTED_CONFIG_KEYS and InferenceEngine::Metrics::METRIC_SUPPORTED_METRICS? So only ov::model_name is necessary?

Yes, I think it is quite strange introduce support of something what should be removed. I propose to add `ov::model_name` only.

File: src/plugins/intel_gna/src/gna_executable_network.hpp [hunk 2]
```diff
@@ -130,6 +132,8 @@ class GNAExecutableNetwork : public InferenceEngine::IExecutableNetworkInternal
     InferenceEngine::Parameter GetMetric(const std::string& name) const override {
         if (ov::supported_properties == name) {
             return Config::GetSupportedProperties(true);
+        } else if (ov::model_name == name) {
+            return networkName;
         } else {
             return plg->GetMetric(name, {});
```
Comment (kbruniec) [inline src/plugins/intel_gna/src/gna_executable_network.hpp:138]: `model_name` can be returned from 
`plg->GetMetric(name, {})`
as other metrics to make it more consistent

Comment (rjeziers) [reply src/plugins/intel_gna/src/gna_executable_network.hpp]: Good idea - done

**Review by nkogteva — APPROVED** [review]

**Review by sirzabek — APPROVED** [review]

**Pull request merged**